src/linear_algebra/pi.lean

```diff
@@ -260,6 +261,22 @@ lemma Pi_congr_right_trans (e : Π i, φ i ≃ₗ[R] ψ i) (f : Π i, ψ i ≃
   (Pi_congr_right e).trans (Pi_congr_right f) = (Pi_congr_right $ λ i, (e i).trans (f i)) :=
 rfl
 
+variables (R φ)
+
+/-- Transport dependent functions through an equivalence of the base space.
+
+This is `equiv.Pi_congr_left'` as a `linear_equiv`. -/
+@[simps {simp_rhs := tt}]
+def Pi_congr_left' (e : ι ≃ ι') : (Π i', φ i') ≃ₗ[R] (Π i, φ $ e.symm i) :=
+{ map_add' := λ x y, rfl, map_smul' := λ x y, rfl, .. equiv.Pi_congr_left' φ e }
+
+/-- Transporting dependent functions through an equivalence of the base,
+expressed as a "simplification".
+
+This is `equiv.Pi_congr_left` as a `linear_equiv` -/
+def Pi_congr_left (e : ι' ≃ ι) : (Π i', φ (e i')) ≃ₗ[R] (Π i, φ i) :=
```

Comment:
Why not `simps` this definition? (And probably even add `simp` lemmas for `refl`, `symm` and `trans`, like for `Pi_congr_right`?)
```suggestion
@[simps]
def Pi_congr_left (e : ι' ≃ ι) : (Π i', φ (e i')) ≃ₗ[R] (Π i, φ i) :=
```

Reply:
I was copying `equiv.Pi_congr_left`, which also doesn't have `simps`. I think just plain `simp` would probably work too.

I didn't add `refl` / `symm` / `trans` because the dependent types make them hard to express and consequently not very useful.

Reply:
For example, the refl version would be something like
```suggestion
def Pi_congr_left_refl : Pi_congr_left R φ (equiv.refl ι) = linear_equiv.cast _ :=
```
but linear_equiv.cast doesn't and probably can't exist.